Expanding the documentation with an outlook of the MESSAGEix usage

LauWien opened the pull request:
This PR creates the initial structure for expanding the documentation by introducing a section to share information about publications, tools, and projects. It serves as a basis for the discussion on how best to approach #519. 
Closes https://github.com/iiasa/message_ix/issues/400.
Closes https://github.com/iiasa/message_ix/issues/519.

## How to review

- Build the documentation and look at main page and linked sub-pages under "Information on tools, projects and publications"

## PR checklist

- [x] Continuous integration checks all ✅

khaeru commented on doc/publications/cooling.rst:
```diff
@@ -0,0 +1,14 @@
+   :width: 250px
+   :align: right
+
+**Hunt, J.D., Weber, N. de A. B., Zakeri, B. et al. (2021), Sustainable Cities and Society**
```
Since we already use [sphinxcontrib-bibtex](https://sphinxcontrib-bibtex.readthedocs.io/en/latest/usage.html), we should use its features and avoid manually formatting citations.

khaeru commented on doc/publications/cooling.rst:
```diff
@@ -0,0 +1,14 @@
+Cooling
+=======
+
+Deep seawater cooling and desalination: Combining seawater air conditioning and desalination
```
Is this the figure caption? If so, see [here](https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure) for how it should be included in the `.. figure:` directive.

LauWien replied:
This was meant to be the heading. The figures don't have captions so far.

khaeru commented on doc/sharing.rst:
```diff
@@ -0,0 +1,161 @@
+Sharing tools, projects and publications
```
Let's not duplicate all this content. Simply reference the existing contributors guide and expand it if necessary.

khaeru commented:
Added some comments inline. We might consider to show the current draft, rendered, as part of code-and-tell at an upcoming meeting, and take suggestions on how it should be structured and phrased.

LauWien commented:
> 
> 
> Added some comments inline. We might consider to show the current draft, rendered, as part of code-and-tell at an upcoming meeting, and take suggestions on how it should be structured and phrased.

Thank you for your review!
We showed the rendered draft once at the 13th of October MESSAGEix meeting and gathered feedback which I have included mainly. But it's good to show the updated version again with your suggestion now and also once I implemented the tool d2ix as a template as @behnam-zakeri suggested. This also helps me to detect feedback I might have overlooked :)

khaeru reviewed (approved):
Thanks for all the work assembling this and pulling it together! Showing the keywords for each publication is a simple and elegant approach and should let readers find what they need.

I rebased and implemented some edits as new commits on the branch—see commit message and comments.

Some additional explanation
- I think an additional subsection on the left navbar is extraneous for a single page. This probable made sense when there were three separate pages; less so now that we've consolidated.
- `doc/index.rst` is really, as the name implies, and index or table of contents. Rather than have paragraphs that appear there and are then duplicated on certain pages, we move them to the pages directly.
- Tried to use the order "publications, projects, tools" in the same order everywhere.

LauWien commented:
> Thanks for all the work assembling this and pulling it together! Showing the keywords for each publication is a simple and elegant approach and should let readers find what they need.
> 
> I rebased and implemented some edits as new commits on the branch—see commit message and comments.
> 
> Some additional explanation
> 
>     * I think an additional subsection on the left navbar is extraneous for a single page. This probable made sense when there were three separate pages; less so now that we've consolidated.
> 
>     * `doc/index.rst` is really, as the name implies, and index or table of contents. Rather than have paragraphs that appear there and are then duplicated on certain pages, we move them to the pages directly.
> 
>     * Tried to use the order "publications, projects, tools" in the same order everywhere.

Thank you for your review and the additional work you put into the PR!
Next time I remember to rebase before asking for the review - thank you for doing that as well!

LauWien commented:
Extended additional the release notes and would squash and merge, what do you think @khaeru?  

Just as FYI:
I have committed the extensions of the release notes, after merging `main` into `issue/519` in GitHub's UI and not before rebasing with `main`. This is because I thought squashing and merging this PR might be better than a merge commit due to the not so helpful commit messages like "_Implement suggestions..._" and it wouldn't then make a difference. Looking now again at the commit messages I'm not so sure anymore, it might make sense to keep them. No matter what I should have rebased :sweat:..

khaeru commented:
> would squash and merge, what do you think @khaeru?

Please do!
